corehq/messaging/scheduling/view_helpers.py

```diff
@@ -94,7 +95,12 @@ def get_worksheet_errors(self, worksheet):
     def upload(self, workbook):
         self.msgs = []
         success_count = 0
-        worksheet = workbook.get_worksheet(title=self.sheet_name)
+
+        try:
+            worksheet = workbook.get_worksheet(title=self.sheet_name)
+        except WorksheetNotFound:
+            return [(messages.error, _("This file is missing the '{sheet_name}' sheet. Please add this sheet "
+                                       "and upload again.".format(sheet_name=self.sheet_name)))]
```

Comment:
Thinking how this would add the error message for a tab while the other one is successfully uploaded and if "upload again" would cause confusion?
How about making it a warning instead of an error?
```
Sheet '{sheet_name}' was not found in the upload.
```

Reply:
Re-uploading again won't hurt, HQ will skip everything in the first tab. Might as well give clear instructions.

Reply:
Hmm, i definitely saw that the first sheet was processed in such a scenario last week. can you help confirm?

Reply:
It should recognize if there are any changes. So if you upload a workbook with only the first sheet, it'll process the first sheet and throw and error that the second is missing. Then if you add the second sheet back in and re-upload, it'll read the first sheet but won't find anything to do, and it'll process the second (new) sheet. Did you see different behavior?

Reply:
that is correct, it was the first upload i was talking about
1. currently there was a 500 which we are addressing now
2. now the user will see a success message for first sheet and error message for the second, which i am saying is confusing.

Reply:
Hmm, the main alternative I can think is to fail the entire upload if any sheet is missing. Do you think that'd be better?

Reply:
hmm, that was akshita's view too. she said "It should be an atomic process".
I leave that decision to you. I am actually fine with this as well, its like app translations upload where we process things per tab, my intention was just to let user know what exactly happened.
(this thread is getting long so just reminding that this is not a blocker :) )

Reply:
Okay, going to leave this as-is for now.